react/components/form/ArrayFieldTemplateItem.js

```diff
@@ -28,9 +28,7 @@ class ArrayFieldTemplateItem extends Component {
     showDragHandle: PropTypes.bool,
   }
 
-  state = {
-    autoHeight: false,
-  }
+  autoHeight = false
```

Comment:
Why did you decide to take `autoHeight` out of the state? It seems to me that it should be there.

Reply:
Inside state, it was provoking infinite loop. Because the component re-renders and the function is called again and again... Outside the state, the behavior is the same, but the component doesn't re-render (better for performance too).

Reply:
But the component is supposed to re-render. You're updating the state in order for it to understand that something should be changed — and re-rendered —, which, in this case, is the style; more specifically the `height`. Moving that logic from the state to a class attribute doesn't seem like a good idea to me, because React supposedly handles the component state and renders efficiently. I think the following code works as desired, could you please test it?

```js
  constructor(props) {
    super(props)

    this.state = {
      autoHeight: false,
    }
  }
```

```js
  handleOnRest = () => {
    if (this.state.autoHeight === false) {
      this.setState({ autoHeight: true })
    }
  }

  handleOnStart = () => {
    if (this.state.autoHeight === true) {
      this.setState({ autoHeight: false })
    }
  }
```

```js
  renderChildren = styles => {
    const { children } = this.props
    return (
      <div
        style={{
          ...styles,
          height: this.state.autoHeight ? 'auto' : styles.height,
        }}
      >
        {children}
      </div>
    )
  }
```

Reply:
It worked! The loop was caused because `handleOnStart` was called right after update state. So it was called again and again. The verification made it work.